target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac

```diff
@@ -25,7 +25,8 @@ case "$board" in
 		[ "$PHYNBR" = "1" ] && \
 			macaddr_add "$(mtd_get_mac_binary factory 0x4)" 1 > /sys${DEVPATH}/macaddress
 		;;
-	jcg,q20)
+	jcg,q20|\
+	yuncore,ax820)
 		[ "$PHYNBR" = "1" ] && \
```

Comment:
You should comment in your commit message on how MAC addresses for WiFi are set (as it seems you have 2g/5g with one radio here?)

Reply:
Thanks, I've updated the commit message. Even though the 2.4GHz/phy0 MAC is the label MAC address, 5GHz is not even set by the factory firmware and randomly generated.

Reply:
Only setting the local bit on the second PHY's MAC address and changing nothing else causes MAC address conflicts when adding multiple VIFs per PHY.
This can be fixed by also incrementing the last byte like this:
https://pastebin.com/CgyLStnf

Reply:
Thanks, took me a while to realize that this won't collide with `lan` because of the local bit.
I've updated the PR in the same code style that's used a few lines above for `dlink,dir-853-r1`: https://github.com/openwrt/openwrt/blob/f12413744f821e8219414e2ef3cff93ead263f41/target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac#L28-L33

Reply:
Well, if vendor firmware uses a random address, we should do so as well. Using the increment will result in having MAC address collisions between different devices in certain cases. (Devices with subsequent MAC addresses do happen in real life.)

Reply:
@adschm the resulting MAC (before setting the local bit) is already used by `lan` on the same device, so there should be no overlaps with other devices.
The phy1 MAC of the factory firmware did not seem to be derived from the label mac, but I'm not 100% sure if it changes at every reboot. I might be able to check this in a few days. but imho BSSIDs should not change at each reboot.

Reply:
(for reference) full explanation here: https://github.com/openwrt/openwrt/pull/4390#issuecomment-1102944208